include/stl2/detail/memory/concepts.hpp

```diff
@@ -18,30 +18,41 @@
 #include <stl2/detail/range/concepts.hpp>
 
 STL2_OPEN_NAMESPACE {
+	///////////////////////////////////////////////////////////////////////////
+	// __NoThrowInputIterator [Exposition]
+	//
+	template <class I>
+	concept bool __NoThrowInputIterator =
+		InputIterator<I> &&
+		_Is<reference_t<I>, std::is_lvalue_reference> &&
+		Same<__uncvref<reference_t<I>>, value_type_t<I>>;
+		// Axiom: no exceptions are thrown from increment, copy, move, assignment, comparison,
+		//        indirection through valid iterators, or comparisons with valid sentinels.
```

Comment:
"comparisons with valid sentinels" is similarly odd since (a) there's no sentinel in sight, and (b) we don't really care if comparison with some value of some type 'S' for which `Sentinel<S,I>` is satisfied can throw if that value and/or type are never passed to one of the algorithms.

Maybe this concept should include the sentinel a la:
```c++
template <class S, class I>
concept bool __NoThrowInputSentinel =
    Sentinel<S, I> && InputIterator<I> && /* ... */;
```